File: code88.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        i=0
        j=0
        while j<n:
            if i>=m+j or nums2[j]<nums1[i]:
                del nums1[-1]
                nums1.insert(i,nums2[j])
                j+=1
            i+=1
        print(nums1)
s=Solution()
s.merge([1,2,3,0,0,0],3,[2,5,6],3)

File: code88.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    void merge(vector<int> &nums1, int m, vector<int> &nums2, int n)
    {
        int id1 = m - 1;
        int id2 = n - 1;
        for (int i = m + n - 1; i >= 0; i--)
        {
            int v1 = INT_MIN;
            int v2 = INT_MIN;
            if (id1 >= 0)
                v1 = nums1[id1];
            if (id2 >= 0)
                v2 = nums2[id2];

            int mx = max(v1, v2);
            if (v1 == mx)
                id1--;
            else if (v2 == mx)
                id2--;
            nums1[i] = mx;
        }
    }
};